cps/kni.c

```diff
@@ -384,69 +384,45 @@ attr_get(struct cps_config *cps_conf,
 		if (family == AF_INET6) {
 			struct in6_addr *addr =
 				mnl_attr_get_payload(tb[RTA_DST]);
+			rte_memcpy(&update->ip.v6,
+				addr, sizeof(update->ip.v6));
 			if (cps_conf->debug)
 				inet6_print("dst", addr);
 		} else {
 			struct in_addr *addr =
 				mnl_attr_get_payload(tb[RTA_DST]);
+			rte_memcpy(&update->ip.v4,
+				addr, sizeof(update->ip.v4));
 			if (cps_conf->debug)
 				inet_print("dst", addr);
 		}
-	}
-	if (tb[RTA_SRC]) {
-		if (family == AF_INET6) {
-			struct in6_addr *addr =
-				mnl_attr_get_payload(tb[RTA_SRC]);
-			if (cps_conf->debug)
-				inet6_print("src", addr);
-		} else {
-			struct in_addr *addr =
-				mnl_attr_get_payload(tb[RTA_SRC]);
-			if (cps_conf->debug)
-				inet_print("src", addr);
-		}
-	}
-	if (tb[RTA_OIF]) {
```

Comment:
Since we send all the packets out through either the front interface or back interface with bonded ports, do we still need to consider about the output interface here?

Reply:
You should notice that `RTA_OIF` is pointing to one of the KNI interfaces, not the hardware interfaces that Gatekeeper uses. Right now, Gatekeeper decides the interface (i.e. `GK_FWD_GATEWAY_FRONT_NET` and `GK_FWD_GATEWAY_BACK_NET`) checking where the gateway is a neighbor. Knowing which KNI interface is the output interface, allows us to check that Gatekeeper is configured correctly and generate warnings log if it's not.

Reply:
This makes sense.